site/content/contribute/server/feature-flags.md

```diff
@@ -0,0 +1,60 @@
+---
+title: "Feature Flags"
+date: 2020-10-15T16:00:00-0700
+weight: 3
+subsection: Server
+---
+
+
+Feature flags allow us to be more confident in shipping features continuously to Mattermost Cloud. Feature flags allow us to control which features are enabled on a cluster level.
+
+# How to use
+
+## Adding the feature flag in code
+
+1. Add the new flag to the feature flag struct located in `model/feature_flags.go`.
+2. Set a default value in the `SetDefaults` function in the same file.
+3. Use the feature flag in code as you would use a regular configuration setting. In tests, manipulate the configuration value to test value changes, such as activation and deactivation, of the feature flag.
+4. Code may be merged regardless of setup in the management system. In this case it will always take the default value supplied in the `SetDefaults` function.
+5. Create a removal ticket for the feature flag. All feature flags should be removed as soon as they have been verified by Cloud. The ticket should encompass removal of the supporting code and archiving in the management system.
+
+### Feature flag code guidelines
+
+- A ticket should be created when a feature flag is added to remove the feature flag as soon as it isn't required anymore.
+- Tests should be written to verify the feature flag works as expected. Note that in cases where there may be a migration or new data, off to on and on to off should both be tested.
+- Log messages by the feature should include the feature flag tag, with the feature flag name as a value, in order to ease debugging.
+
+
+## Deploying to split.io
+
+Deployments to the managment system are overseen by the Cloud team. If you have any questions or need help with the process, please ask in the cloud channel.
+
+1. When ready to deploy the feature, create the feature flag (called a split) in split.io. The name of the feature flag should match the name of the split. Everything else can be left at defaults.
+2. Once created, set the treatment values appropriately. The defaults of "on" an "off" can work for most feature flags.
+3. When ready to deploy to cloud, set the targeting rules appropriately to slowly roll out as required. 
+
+## Timelines for rollouts
+
+< Guidelines coming soon > 
```

Comment:
1. The feature is tested on a PR and E2E test automation is added. The feature is added behind a feature flag and can be turned on or off as needed.
2. [T-8] Prior to the cut to on-premise, the Release manager reviews new feature flags and all story tickets from cloud releases since the previous on-premise release and provides a report to PM/QA team for their review. PM/QA review stories and features flags to determine if any should not be included in the on-premise release based on the following criteria:
 - The feature has not been available to cloud instances for at least 2 weeks.
 - The feature has 1 or more severe bugs.
 - The feature has a high amount of outstanding bug tickets (5 or more), unless they are very superficial or minor issues.
 - The feature is considered “experimental” and is still in active development.
3. Any feature that should be changed or omitted from the on-premise release is logged via a Jira Ticket by the PM team for their respective development team to exclude or change the feature before on-premise release branch is cut.
4. [T-7] On-premise release branch and release candidate are cut based on cloud release.

Reply:
Chatted offine. Added note for on-prem releases.